### PR TITLE
AirbyteLib: Add support for JSON and VARIANT types

### LINKED ISSUE #25: 🐛 Bug: Object types not supported in Snowflake and Postgres caches
Currently, columns of type 'object' will load successfully in DuckDB but not in Snowflake/Postgres.

Work started here:

- https://github.com/airbytehq/airbyte/pull/35117

### PR DESCRIPTION
**Pull request by aaronsteers**
Resolves: https://github.com/airbytehq/airbyte-lib-private-beta/issues/25

This PR adds support for JSON and VARIANT types.

To do this, I went ahead and built a Snowflake-native implementation of `_write_files_to_new_table()`. The implementation now uses `PUT` to upload files to Snowflake table stages, then runs `COPY` to get the files' data into the table.

It is a carry-on from Joe's PR:

- #34942

### CODE REVIEW

File: airbyte-lib/airbyte_lib/caches/base.py [hunk 1]
```diff
@@ -809,13 +809,14 @@ def _swap_temp_table_with_final_table(
...
-        for cmd in commands:
-            self._execute_sql(cmd)
+        commands = "\n".join(
+            [
+                f"ALTER TABLE {final_table_name} RENAME TO {deletion_name};",
+                f"ALTER TABLE {temp_table_name} RENAME TO {final_table_name};",
+                f"DROP TABLE {deletion_name};",
+            ]
+        )
+        self._execute_sql(commands)
```
Comment (aaronsteers) [inline airbyte-lib/airbyte_lib/caches/base.py:819]: Slight speed boost by sending all three commands at once, rather then having three round trips.

File: airbyte-lib/airbyte_lib/types.py [hunk 2]
```diff
@@ -22,8 +22,8 @@
     "time_without_timezone": sqlalchemy.types.TIME,
     # Technically 'object' and 'array' as JSON Schema types, not airbyte types.
     # We include them here for completeness.
-    "object": sqlalchemy.types.VARCHAR,
```
Comment (bindipankhudi) [inline airbyte-lib/airbyte_lib/types.py:25]: just for me to understand, why don't we want the varchar anymore?

File: airbyte-lib/tests/integration_tests/test_snowflake_cache.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    """Test that the append strategy works as expected."""
+    result = source_faker_seed_a.read(
+        snowflake_cache, write_strategy="replace", force_full_refresh=True
+    )
+    assert len(list(result.cache.streams["users"])) == FAKER_SCALE_A
+
+
+def test_replace_strategy(
+    source_faker_seed_a: ab.Source,
+    snowflake_cache: ab.DuckDBCache,
```
Comment (bindipankhudi) [inline airbyte-lib/tests/integration_tests/test_snowflake_cache.py:114]: nit. should the parameter name be duckDB_cache ?

**Review by bindipankhudi — APPROVED** [review]
Just a minor comment on integration tests. nothing blocking,

File: airbyte-lib/tests/integration_tests/test_snowflake_cache.py (hunk 3 above)
Comment (aaronsteers) [reply airbyte-lib/tests/integration_tests/test_snowflake_cache.py:114]: Good catch! Thanks!

```suggestion
    snowflake_cache: ab.SnowflakeCache,
```

**Review by bindipankhudi — APPROVED** [review]
looks good!

File: airbyte-lib/airbyte_lib/types.py (hunk 2 above)
Comment (aaronsteers) [reply airbyte-lib/airbyte_lib/types.py:25]: It's actually not being removed, but the diff is displaying strangely. It should render as "object" type having its mapping changed from `sqlalchemy.types.VARCHAR` to `sqlalchemy.types.JSON`

**Pull request merged**